src/Morpho.sol

```diff
@@ -160,7 +160,7 @@ contract Morpho is IMorphoStaticTyping {
         emit EventsLib.CreateMarket(id, marketParams);
 
         // Call to initialize the IRM in case it is stateful.
-        IIrm(marketParams.irm).borrowRate(marketParams, market[id]);
+        if (marketParams.irm != address(0)) IIrm(marketParams.irm).borrowRate(marketParams, market[id]);
```

Comment:
From my understanding, this PR is to allow the creation of "idle" markets where there's no IRM. Is my understanding correct?

This allows anyone to create a market where no interest will accrue.

- borrowers do not need to pay any interest to suppliers
- suppliers do not accrue and earn any earning from borrowers
- bad debt socialization can't be "neutralized" by interest accrued in the past

Isn't there any better solution to allow the creation of "idle" markets? Because with this modification and the fact that you are not checking during market creation if the `loanToken` and `collateralToken` exists (`!= address(0)` or `address(token).code.length > 0`) it could lead to some unwanted behaviors.

Given that you are shaping MB to allow the MM use case (idle markets), I would suggest thinking about:

1) find the conditions that **only** allow the creation of an idle market (supply `loanToken` without allowing borrowing, only withdraw)
2) does not allow the creation of arbitrary markets that do no make sense for a landing platform

Reply:
> Is my understanding correct?

Correct

> As I mentioned previously, IMHO, Morpho blue should not be designed and shaped for the needs of externals actors like MetaMorpho. MetaMorpho should be shaped around the basic building block that MB should be.

Indeed, it's 100% aligned with our general philosophy but we think it makes more sense from a product perspective and we won't change it.

Reply:
>This allows anyone to create a market where no interest will accrue.
>
>borrowers do not need to pay any interest to suppliers
suppliers do not accrue and earn any earning from borrowers
bad debt socialization can't be "neutralized" by interest accrued in the past

correct, but this is not the expected use-case: for an idle market, one would put address zero as collateral and no borrower would be able to deposit collateral (and so borrow). It's only a way to make these idle depositors pay less gas.

Reply:
I'm confused because this change has been done specifically to allow the creation of "idle market" but it also allows the creation of unwanted (as far as I get) markets.

Why don't you narrow the creation of those market to just the one that makes sense to be used as an idle market (as you said)?

Reply:
agree that it feels weird, but in Blue you can also created unwanted markets in multiple ways (tokens are not checked, oracles are not checked) so why checking for this thing specifically

Reply:
> agree that it feels weird, but in Blue you can also created unwanted markets in multiple ways (tokens are not checked, oracles are not checked) so why checking for this thing specifically

Because this modification has been made deliberately to allow the creation of an idle market, with the side effects that it allows the creation of non-lending markets. 

The final decision is obviously up to you, I just wanted to point out the side effects and that at least those side effects need to be documented.